companydata/gap-sogessur-emit.go

```diff
@@ -0,0 +1,467 @@
+		}
+		slice[index] = value
+	}
+	return slice, nil
+}
+
+func getGapHeader() []string {
+	return []string{
+		"NUMERO POLIZZA",
+		"NUMERO CONTRATTO",
+		"TIPO OPERAZIONE",
+		"DATA OPERAZIONE",
+		"COGNOME/RAGIONE SOCIALE ASSICURATO",
+		"NOME ASSICURATO",
+		"INDIRIZZO ASSICURATO",
+		"COMUNE ASSICURATO",
+		"CAP ASSICURATO",
+		"PROVINCIA ASSICURATO",
+		"NAZIONE ASSICURATO",
+		"CODICE FISCALE ASSICURATO",
+		"PARTITA IVA ASSICURATO",
+		"TARGA",
+		"TELAIO",
+		"MODELLO",
+		"MARCA",
+		"CILINDRATA",
+		"ALIMENTAZIONE",
+		"ANTIFURTO SATELLITARE",
+		"TIPO VEICOLO",
+		"TIPO TARGA",
+		"CAVALLI",
+		"QUINTALI",
+		"KW",
+		"CODICE MODELLO",
+		"VEICOLO NUOVO",
+		"DATA CONSEGNA",
+		"DATA IMMATRICOLAZIONE",
+		"CODICE DEALER",
+		"DEALER",
+		"PACCHETTO",
+		"CAPITALE ASSICURATO",
+		"DATA INIZIO POLIZZA",
+		"DATA FINE POLIZZA",
+		"ORA EFFETTO",
+		"ORA SCADENZA",
+		"DATA SCADENZA RATE",
+		"TIPO FRAZIONAMENTO",
+		"NUMERO RATE",
+		"DURATA COPERTURA",
+		"PREMIO NETTO GAP",
+		"IMPOSTE GAP",
+		"PREMIO LORDO GAP",
+		"PREMIO NETTO CVT",
+		"IMPOSTE CVT (comprensivo dell'1%)",
+		"PREMIO LORDO CVT",
+		"PREMIO NETTO PAI",
+		"IMPOSTE PAI",
+		"PREMIO LORDO PAI",
+		"PREMIO NETTO TL",
+		"IMPOSTE TL",
+		"PREMIO LORDO TL",
+		"PROVVIGIONI GAP",
+		"PROVVIGIONI PAI",
+		"PROVVIGIONI TL",
+		"PROVVIGIONE BK",
+		"PROVVIGIONE BK PAI",
+		"PROVVIGIONE BK TL",
+		"FEE MGT",
+		"TIPO PACCHETTO",
+		"IBAN",
+		"DATA INCASSO",
+		"VINCOLO",
+		"DATA VINCOLO",
+		"ENTE VINCOLATARIO",
+		"CODICE ZONA",
+		"CL_SESSO",
+		"CL_DATA_NASC",
+		"CL_LUOGO_NASC",
+		"CL_PROV_NASC",
+		"COGNOME/RAGIONE SOCIALE PROPRIETARIO",
+		"NOME PROPRIETARIO",
+		"INDIRIZZO PROPRIETARIO",
+		"COMUNE PROPRIETARIO",
+		"CAP PROPRIETARIO",
+		"PROVINCIA PROPRIETARIO",
+		"NAZIONE PROPRIETARIO",
+		"CODICE FISCALE PROPRIETARIO",
+		"PARTITA IVA PROPRIETARIO",
+		"PR_SESSO",
+		"PR_DATA_NASC",
+		"PR_LUOGO_NASC",
+		"PR_PROV_NASC",
+	}
+}
+
+func getGapRowMap(policy models.Policy, transaction models.Transaction) map[string]string {
+	vehicle := policy.Assets[0].Vehicle
+	contractor := policy.Contractor
+	vehicleOwner := policy.Assets[0].Person
+	offerName := policy.OfferlName
+
+	genders := []string{"F", "M", "G"} // For validation
+	// Maps for replacing the values in policy into valid values for the gap csv
+	boolAnswers := map[bool]string{
+		true:  "SI",
+		false: "NO",
+	}
+	gapOptions := map[string]string{
+		"base":     "OPTION 1",
+		"complete": "OPTION 2",
+	}
+	isVehicleNew := map[string]string{
+		"Nuovo": "SI",
+		"Usato": "NO",
+	}
+	powerSupplyCodes := map[string]string{
+		"Sì": "E",
+		"No": "",
+	}
+	// Assuming we have only this payment type
+	offer := policy.OffersPrices[offerName][string(models.PaySingleInstallment)]
+	vehicleWeight := ""
+	if vehicle.Weight > 0 {
+		vehicleWeight = strconv.Itoa(int(vehicle.Weight))
+	}
+	zoneCode := quote.GetAreaByProvince(vehicleOwner.Residence.CityCode)
```

Comment:
save this info inside `Address` struct in a field name `Area` of type string and remove dependency to `quote`